C7/PCXToGodot.cs

```diff
@@ -5,65 +5,102 @@
 public class PCXToGodot : Godot.Object
 {
 	private readonly static byte CIV3_TRANSPARENCY_START = 254;
-	
-	public static ImageTexture getImageTextureFromPCX(Pcx pcx) {
-		Image ImgTxtr = ByteArrayToImage(pcx.ColorIndices, pcx.Palette, pcx.Width, pcx.Height);
+
+	public static ImageTexture getImageTextureFromImage(Image image) {
 		ImageTexture Txtr = new ImageTexture();
-		Txtr.CreateFromImage(ImgTxtr, 0);
+		Txtr.CreateFromImage(image, 0);
 		return Txtr;
 	}
 
+	public static ImageTexture getImageTextureFromPCX(Pcx pcx) {
+		Image ImgTxtr = ByteArrayToImage(pcx.ColorIndices, pcx.Palette, pcx.Width, pcx.Height);
+		return getImageTextureFromImage(ImgTxtr);
+	}
+
 	public static ImageTexture getImageTextureFromPCX(Pcx pcx, int leftStart, int topStart, int croppedWidth, int croppedHeight) {
-		Image Image = getImageFromPCX(pcx, leftStart, topStart, croppedWidth, croppedHeight);
-		ImageTexture Txtr = new ImageTexture();
-		Txtr.CreateFromImage(Image, 0);
-		return Txtr;
+		Image image = getImageFromPCX(pcx, leftStart, topStart, croppedWidth, croppedHeight);
+		return getImageTextureFromImage(image);
+	}
+
+	// BufferData should be as large as the largest PCX file needed to be loaded in pixels
+	// For now, I will assume no PCX file is larger than 4k resolution
+	// It's efficient to use a shared BufferData, ColorData, and AlphaData instead of having to reinitialize these arrays for every Image load
+	// However, this will become a problem if (when?) we move towards parallelization
+	// But for now, it's an improvement
+	public static int[] BufferData = new int[3840 * 2160];
+	public static int[] ColorData = new int[256];
+	public static int[] AlphaData = new int[256];
+
+	public static void loadPalette(byte[,] palette, bool shadows = false) {
```

Comment:
Same as above: this should probably be private.  That makes it clear there's no need to check for external references.

Reply:
Same with loadAlphaPalette, getImageFromBufferData.

As a reviewer, this also helps make it clear what the top level is.  If it's new and it's private, I know that it's a helper method and I shouldn't start by reviewing that method until I know how it's used at a higher level.

I also generally prefer that helper methods be below the methods that call them, for the same basic reason.  Without guidance, I generally explore files top-to-bottom, with a preference for starting at the entry points (public/protected methods).  Back in the old days (C89, for instance), helper methods had to be defined first so the compiler could find them, but nowadays I prefer human readability.

Reply:
Ah, I tend to default to the C++ ways of writing callees before callers (because I'm too lazy to write function declarations), but I'm fine if this is the preferred style here.

Reply:
Honestly, one of the really nice things about Java and C# is not having to write function declarations.  It's a small thing, but it's one of the things I have to remind myself about when I go back and write some C.  Taking a look at some of my 2019 C code, sure enough, the callees are first, probably because I didn't want to have to write a .h file/include more definitions in the .h file.

It's also kind of a combined-effect thing.  I may well not have commented on the order if the public interfaces were the only public things, and if it had been in top-down order, I may have overlooked that the local methods weren't private reading top-down.